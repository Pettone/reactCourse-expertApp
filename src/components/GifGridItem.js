import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, title, url} ) => {

    console.log({id,title,url});

    return (
        <div className = "card animate__animated animate__bounce animate__delay-2s">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
