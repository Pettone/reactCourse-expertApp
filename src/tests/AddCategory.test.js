import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../components/AddCategory';

describe('Tests in AddCategory component', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('Must to show itself correctly', () => { 

        expect(wrapper).toMatchSnapshot();
    });

    test('Must to change the text box', () => { 

        const input = wrapper.find('input');
        const value = 'I am a value';

        input.simulate('change', { target: { value } } );
    });

    test('Do not post information with submit', () => { 

        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Must to call setCategories and clean box text', () => {

        const value = 'Hi World';
        // Simulating inputChange...
        wrapper.find('input').simulate('change', { target: {value} });

        // Simulating submit...
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        // setCategories must to have been called...
        expect(setCategories).toHaveBeenCalled();

        // input value must to be a empty string...
        expect( wrapper.find('input').prop('value') ).toBe('');
    });
});