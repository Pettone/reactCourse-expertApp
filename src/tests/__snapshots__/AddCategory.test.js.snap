// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in AddCategory component Must to show itself correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
